File: 0137-single-number-ii/0137-single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int ans = 0;
        for (int i = 0; i < 32; i++)
        {
            int setBit = 0;
            for (auto num: nums)
            {
                if (num & (1 << i))
                {
                    setBit += 1;
                }
            }
            if (setBit % 3 == 1)
            {
                ans |= (1 << i);
            }
        }

        return ans;
    }
};

File: 0137-single-number-ii/0137-single-number-ii.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        ans = 0
        for i in range(32):
            setBit = 0
            for num in nums:
                if num & (1 << i) != 0:
                    setBit += 1
            if setBit % 3 == 1:
                ans = ans | (1 << i)
        
        return ans - (1 << 32) if ans >= (1 << 31) else ans